rustic-cargo.el

```diff
@@ -349,30 +357,42 @@ Execute process in PATH."
   (interactive)
   (tabulated-list-put-tag " " t))
 
+(cl-defstruct rustic-crate name version)
+
+(defun rustic-cargo--outdated-make-crate (crate-line)
+  "Create RUSTIC-CRATE struct out of a CRATE-LINE.
+
+The CREATE-LINE is a single line from the `rustic-cargo-oudated-buffer-name'"
+  (make-rustic-crate :name (nth 1 crate-line) :version (nth 2 crate-line)))
+
 ;;;###autoload
 (defun rustic-cargo-upgrade-execute ()
   "Perform marked menu actions."
   (interactive)
-  (let (crates)
-    (save-excursion
-      (goto-char (point-min))
-      (while (not (eobp))
-        (let* ((cmd (char-after))
-               (crate (tabulated-list-get-entry (point))))
-          (when (eq cmd ?U)
-            (push crate crates)))
-        (forward-line)))
+  (let ((crates (rustic-cargo--outdated-get-crates (buffer-string))))
     (if crates
-        (let ((msg (format "Upgrade %s ?" (mapconcat #'(lambda (x) (elt x 0)) crates " "))))
+        (let ((msg (format "Upgrade %s ?" (mapconcat #'(lambda (x) (rustic-crate-name x)) crates " "))))
           (when (yes-or-no-p msg)
             (rustic-cargo-upgrade-crates crates)))
       (user-error "No operations specified"))))
 
+(defun rustic-cargo--outdated-get-crates (cargo-outdated-buffer-string)
+  "Return a list of RUSTIC-CRATE which needs to be updated.
```

Comment:
I guess you forgot to update "RUSTIC-CRATE" after you removed the argument.

Reply:
Ok, got it.

Reply:
I made it in backticks. From a quick look into the the EMACS manual, I wasn't able to find/confirm if that's right. But I will further investigate it tomorrow. Thanks!

Reply:
AFAIK you only make functions arguments in upper case, but I'm not 100% sure.

Reply:
checkdoc usually knows what's common practice. But I don't like it as part of the test suite.